<?xml version="1.0"?>
<launch>
    <!--node name="rtabmap" ns="rtab_map" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
        <remap from="/rtab_map/rgb/image" to="/camera/rgb/image_raw"/>
        <remap from="/rtab_map/rgb/camera_info" to="/camera/rgb/camera_info"/>
        <remap from="/rtab_map/depth/image" to="/camera/depth/image"/>
        <remap from="/rtab_map/scan" to="/scan"/>
        <remap from="/rtab_map/grid_map" to="/map"/>
        <param name="subscribe_depth" type="bool" value="true"/>
        <param name="subscribe_scan" type="bool" value="true"/>
        <param name="RGBD/NeighborLinkRefining" type="string" value="true"/>
        <param name="odom_frame_id" value="odom"/>
        <param name="frame_id" value="ROSbot2_base"/>
        <param name="queue_size" value="50"/>
        <param name="RGBD/NeighborLinkRefining" type="string" value="true"/>
        <param name="RGBD/ProximityBySpace" type="string" value="true"/>
        <param name="RGBD/AngularUpdate" type="string" value="0.01"/>
        <param name="RGBD/LinearUpdate" type="string" value="0.01"/>
        <param name="RGBD/OptimizeFromGraphEnd" type="string" value="false"/>
        <param name="Optimizer/Slam2D" type="string" value="true"/>
        <param name="Reg/Strategy" type="string" value="1"/>
        <param name="Reg/Force3DoF" type="string" value="true"/>
        <param name="Vis/MinInliers" type="string" value="5"/>
        <param name="Vis/InlierDistance" type="string" value="0.1"/>
        <param name="Rtabmap/TimeThr" type="string" value="700"/>
        <param name="Mem/RehearsalSimilarity" type="string" value="0.45"/>
    </node todo esto lo he cogido de github de un pavo -->
    
    <group ns="rtabmp">
    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
    
        <!--remap from="/rtab_map/rgb/image" to="/camera/rgb/image_raw"/>
        <remap from="/rtab_map/rgb/camera_info" to="/camera/rgb/camera_info"/>
        <remap from="/rtab_map/depth/image" to="/camera/depth/image"/>
        <remap from="/rtab_map/scan" to="/scan"/>
        <remap from="/rtab_map/grid_map" to="/map"/-->
        
        
        <!-- debes de remapear segun los topicos que tengas bro-->
        <remap from="odom" to="">
        <remap from="scan" to="">  <!--este no se si esta bien puesto debido a que ya hago un remap en el launcher de ptc_to_laserscan-->
        <remap from="rgb/image" to="/camera/rgb/image_color">  <!--/camera/rgb/image_raw-->
        <remap from="depth/image" to="/camera/depth_registered/points">
        <remap from="rgb/camera_info" to="/camera/rgb/camera_info">
        
        
        <param name="frame_id" type="string" value="base_link"/>
        
        <param name="subscribe_depth" type="bool" value="true"/>
        <param name="subscribe_scan" type="bool" value="true"/>
        
        <param name="queue_size" type="int" value="10"/>
        
        <param name="RGBD/NeighborLinkRefining" 	type="string" value="true"/>
        <param name="RGBD/ProximityBySpace" 		type="string" value="true"/>
        <param name="RGBD/AngularUpdate" 		type="string" value="0.01"/>
        <param name="RGBD/LinearUpdate" 		type="string" value="0.01"/>
        <param name="RGBD/OptimizeFromGraphEnd" 	type="string" value="false"/>
        <param name="Optimizer/Slam2D" 			type="string" value="true"/>
        <param name="Reg/Strategy" 			type="string" value="1"/>
        <param name="Reg/Force3DoF" 			type="string" value="true"/>
        <param name="Vis/MinInliers" 			type="string" value="5"/>
        <param name="Vis/InlierDistance" 		type="string" value="0.1"/>
        <param name="Rtabmap/TimeThr" 			type="string" value="700"/>
        <param name="Mem/RehearsalSimilarity" 		type="string" value="0.45"/>
    </node>
    </group>
</launch>
